src/Message/CIMCreateCardRequest.php

```diff
@@ -174,16 +174,27 @@ public function sendData($data)
         if (isset($this->getParameters()['customerProfileId'])) {
             $createPaymentProfileResponse = $this->makeCreatePaymentProfileRequest($this->getParameters());
 
-            if (!$createPaymentProfileResponse->isSuccessful()) {
-                return $createPaymentProfileResponse;
-            }
+            $parameters = [];
 
-            $parameters = array(
-                'customerProfileId' => $createPaymentProfileResponse->getCustomerProfileId(),
-                'customerPaymentProfileId' => $createPaymentProfileResponse->getCustomerPaymentProfileId()
-            );
+            if (!$createPaymentProfileResponse->isSuccessful() && $createPaymentProfileResponse->getReasonCode() === 'E00039') {
+                $data = $createPaymentProfileResponse->getData();
 
-            $response = $this->makeGetPaymentProfileRequest($parameters);
+                if ($data && isset($data['customerPaymentProfileId'])) {
+                    $parameters = [
+                        'customerProfileId' => $this->getParameters()['customerProfileId'],
+                        'customerPaymentProfileId' => $data['customerPaymentProfileId']
+                    ];
+                }
+            } else {
```

Comment:
just do ` } elseif ($response->isSuccessful()) {` here instead? (similar to the block below) that way if you get a non E00039 error you wont enter here.

or for better readability you can do
```
if response succeessful {
   // success logic
} else if ! response succeessful  and error code E00039 {
 // E00039 logic
}